{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "23d94e2e-60be-41fd-aec3-cd9f8c72f2ad",
   "metadata": {},
   "outputs": [],
   "source": [
    "#!/bin/bash\n",
    "#SBATCH -N 1 # number of nodes\n",
    "#SBATCH -n 16 # number of cores\n",
    "#SBATCH --mem=48G\n",
    "#SBATCH --job-name=\"Umar_preprocessing\"\n",
    "#SBATCH --mail-type=END\n",
    "#SBATCH --partition=computeq\n",
    "#SBATCH --mail-user=hrclndnn@memphis.edu\n",
    "#SBATCH -t 14-00:00:00\n",
    "#SBATCH -a 0-81\n",
    "\n",
    "module load bwa\n",
    "module load samtools\n",
    "module load picard\n",
    "module load sratoolkit\n",
    "module load gatk\n",
    "module load bcftools\n",
    "\n",
    "files=(SRR517013\n",
    "SRR518661\n",
    "SRR518662\n",
    "SRR518670\n",
    "SRR518671\n",
    "SRR518672\n",
    "SRR518673\n",
    "SRR518676\n",
    "SRR518677\n",
    "SRR518678\n",
    "SRR518679\n",
    "SRR518680\n",
    "SRR518684\n",
    "SRR518685\n",
    "SRR518686\n",
    "SRR518687\n",
    "SRR518688\n",
    "SRR518689\n",
    "SRR518699\n",
    "SRR518700\n",
    "SRR518701\n",
    "SRR518702\n",
    "SRR518703\n",
    "SRR827537\n",
    "SRR827574\n",
    "SRR827584\n",
    "SRR827585\n",
    "SRR827587\n",
    "SRR827588\n",
    "SRR827600\n",
    "SRR942197\n",
    "SRR942203\n",
    "SRR942211\n",
    "SRR942218\n",
    "SRR942235\n",
    "SRR942239\n",
    "SRR942240\n",
    "SRR942246\n",
    "SRR942248\n",
    "SRR942249\n",
    "SRR942251\n",
    "SRR942253\n",
    "SRR942260\n",
    "SRR942265\n",
    "SRR942266\n",
    "SRR942273\n",
    "SRR942275\n",
    "SRR942276\n",
    "SRR942277\n",
    "SRR942278\n",
    "SRR942279\n",
    "SRR942280\n",
    "SRR942281\n",
    "SRR942282\n",
    "SRR942283\n",
    "SRR942284\n",
    "SRR942285\n",
    "SRR942286\n",
    "SRR942287\n",
    "SRR942288\n",
    "SRR942289\n",
    "SRR942290\n",
    "SRR942291\n",
    "SRR942292\n",
    "SRR942293\n",
    "SRR942294\n",
    "SRR942295\n",
    "SRR942296\n",
    "SRR942297\n",
    "SRR942298\n",
    "SRR942299\n",
    "SRR942300\n",
    "SRR942301\n",
    "SRR942302\n",
    "SRR942303\n",
    "SRR942304\n",
    "SRR942305\n",
    "SRR942306\n",
    "SRR942307\n",
    "SRR942308\n",
    "SRR942309\n",
    "SRR942310\n",
    "SRR947747)\n",
    "\n",
    "file=${files[$SLURM_ARRAY_TASK_ID]}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fe46dd5b-cb66-4c26-a256-35b69210f68d",
   "metadata": {},
   "outputs": [],
   "source": [
    "#00 pull samples from NCBI\n",
    "#cd /home/hrclndnn/sra_local/\n",
    "#prefetch --max-size 100G ${file}\n",
    "#fastq-dump --split-files ${file}\n",
    "\n",
    "#01 Align to reference genome\n",
    "#bwa mem -M -t 16 /home/mdpllard/Genomes/Ursus_maritimus.fna /home/hrclndnn/WGSUmari/00_rethread/${file}_1.fastq /home/hrclndnn/WGSUmari/00_rethread/${file}_2.fastq > /home/hrclndnn/WGSUmari/01_mappedSam/S${file}_mapped.sam\n",
    "\n",
    "#02 Create a sorted bam file from the mapped sam file\n",
    "gatk --java-options '-Djava.io.tmpdir=/home/scratch/hrclndnn/' SortSam -I /home/hrclndnn/WGSUmari/01_mappedSam/${file}_mapped.sam -O /home/hrclndnn/WGSUmari/02_sortedBam/${file}.sorted.bam -SO coordinate "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3821becd-f3b2-4855-95b8-faa5ad058762",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
